### PR TITLE
planner: add Fix52592 control to disallow fast path

### LINKED ISSUE #52592: fast-path should support column projection pushdown
## Feature Request

**Is your feature request related to a problem? Please describe:**

The fast-path point get in TiDB fetch the whole row from TiKV. There is no column projection pushdown for fast-path Get request as Cop result. This can cause performance degradation in below scenarios are:

1. Wide tables with many columns, where applications only query a few columns.
2. JSON fields with large column values, where queries may not necessarily retrieve the values of JSON columns or only extract specific small JSON content. 

**Describe the feature you'd like:**

fast-path point get should support column projection pushdown to TiKV, to reduce the GRPC response message size from TiKV and improve the performance, as well as the resource utilization.
Tn addition, the optimizer need to be aware it's worth to do column projection base on the statistics.

**Describe alternatives you've considered:**

none

**Teachability, Documentation, Adoption, Migration Strategy:**

performance enhancement.

### PR DESCRIPTION
**Pull request by dbsid**
### What problem does this PR solve?

Issue Number: close #52592

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
add doc for Fix52592
```

### CODE REVIEW

File: pkg/planner/core/point_get_plan.go [hunk 1]
```diff
@@ -883,15 +884,17 @@ type PointPlanVal struct {
...
 		// the rule of stabilizing results has not taken effect yet, so cannot generate a plan here in this mode
+		// or Fix52592 is turn on to disable fast path for select, update and delete
 		return nil
 	}
 
 	ctx.GetSessionVars().PlanID.Store(0)
 	ctx.GetSessionVars().PlanColumnID.Store(0)
 	switch x := node.Node.(type) {
 	case *ast.SelectStmt:
+		// TODO: move this up to cover Delete as well after we have finished the fix for https://github.com/pingcap/tidb/issues/38911
```
Comment (winoros) [inline pkg/planner/core/point_get_plan.go:897]: I think we don't need this TODO. The switch should cover all point get behavior.

**Review by winoros — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Pull request merged**